lib/connectors/memory.js

```diff
@@ -352,29 +352,61 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+
   for (var key in data) {
     var val = data[key];
-    if (val === undefined || val === null) {
-      continue;
-    }
+
     if (props[key]) {
```

Comment:
As I mentioned in an earlier [comment](https://github.com/strongloop/loopback-datasource-juggler/pull/1169#discussion_r91296999), please move this check into `_castPropertyValue`.